interfaces/tests/file_int/file_int.c

```diff
@@ -0,0 +1,637 @@
+    }
+    file_destroy(file_handle);
+    delete_file(filename);
+} 
+
+
+/*Tests_SRS_FILE_43_003: [If a file with name full_file_name does not exist, file_create shall create a file with that name.]*/
+/*Tests_SRS_FILE_43_001: [file_create shall open the file named full_file_name for asynchronous operations and return its handle.]*/
+/*Tests_SRS_FILE_43_006: [file_destroy shall wait for all pending I / O operations to complete.]*/
+/*Tests_SRS_FILE_43_007: [file_destroy shall close the file handle handle.]*/
+/*Tests_SRS_FILE_43_014: [file_write_async shall enqueue a write request to write source's content to the position offset in the file. ]*/
+/*Tests_SRS_FILE_43_041: If position + size is greater than the size of the file and the call to write is successfull, file_write_async shall grow the file to accomodate the write.]*/
+/*Tests_SRS_FILE_43_008: [file_write_async shall call user_call_back passing user_context and success depending on the success of the asynchronous write operation.]*/
+/*Tests_SRS_FILE_43_030: [file_write_async shall succeed and return FILE_WRITE_ASYNC_OK.]*/
+/*Tests_SRS_FILE_43_021: [file_read_async shall enqueue a read request to read handle's content at position offset and write it to destination. ]*/
+/*Tests_SRS_FILE_43_016: [file_read_async shall call user_callback passing user_context and success depending on the success of the asynchronous read operation.]*/
+/*Tests_SRS_FILE_43_031: [file_read_async shall succeed and return FILE_READ_ASYNC_OK.]*/
+TEST_FUNCTION(large_simultaneous_reads_succeed)
+{
+    ///arrange
+    int block_size = 4096;
+    int num_blocks = 50;
+    unsigned char* source = (unsigned char*)malloc(block_size*num_blocks);
+    ASSERT_IS_NOT_NULL(source);
+
+    for (int i = 0; i < num_blocks; ++i)
+    {
+        (void)memset(&source[i * block_size], 'a' + i, block_size);
+    }
+
+    WRITE_COMPLETE_CONTEXT write_context;
+    write_context.pre_callback_value = 41;
+    (void)interlocked_exchange(&write_context.value, write_context.pre_callback_value);
+    write_context.post_callback_value = 42;
+
+    char filename[] = "large_simultaneous_reads_succeed.txt";
+    FILE_HANDLE file_handle = file_create_helper(filename);
+
+    ASSERT_ARE_EQUAL(FILE_WRITE_ASYNC_RESULT, FILE_WRITE_ASYNC_OK, file_write_async(file_handle, source, block_size * num_blocks, 0, write_callback, &write_context));
+    wait_on_address_helper(&write_context.value, &write_context.pre_callback_value, UINT32_MAX);
+
+    ASSERT_ARE_EQUAL(int32_t, write_context.post_callback_value, interlocked_or(&write_context.value, 0), "value should be post_callback_value");
+    ASSERT_IS_TRUE(write_context.did_write_succeed);
+
+    READ_COMPLETE_CONTEXT contexts[50];
+    unsigned char* destinations[50];
+
+    ///act
+    for (int i = 0; i < num_blocks; ++i)
+    {
+        destinations[i] = (unsigned char*)malloc(block_size);
+        ASSERT_IS_NOT_NULL(destinations[i]);
+        (void)memset(destinations[i], 'a' + i, block_size);

```

Comment:
>(void)memset(destinations[i], 'a' + i, block_size); [](start = 8, length = 51)

:) Probably not intended here as you want to make sure you read the data (copy pasta suspected)

Reply:
oops, thanks for the catch!